keyboards/crkbd/keymaps/antosha417/keymap.c

```diff
@@ -0,0 +1,429 @@
+
+void oled_render_layer_state(void) {
+    oled_write_P(PSTR("layer: "), false);
+    if (layer_state < L_QWERTY) {
+      oled_write_ln_P(PSTR("qwerty"), false);
+    } else if (layer_state < L_DVORAK) {
+      oled_write_ln_P(PSTR("dvorak"), false);
+    } else if (layer_state < L_LOWER) {
+      oled_write_ln_P(PSTR("lower"), false);
+    } else if (layer_state < L_RAISE) {
+      oled_write_ln_P(PSTR("raise"), false);
+    } else if (layer_state < L_TOP) {
+      oled_write_ln_P(PSTR("top"), false);
+    } else if (layer_state < L_BOTTOM) {
+      oled_write_ln_P(PSTR("bottom"), false);
+    } else {
+      oled_write_ln_P(PSTR("unknown"), false);
+    }
+}
+
+
+char keylog_str[24] = {};
+
+const char code_to_name[60] = {
+    ' ', ' ', ' ', ' ', 'a', 'b', 'c', 'd', 'e', 'f',
+    'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p',
+    'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
+    '1', '2', '3', '4', '5', '6', '7', '8', '9', '0',
+    'R', 'E', 'B', 'T', '_', '-', '=', '[', ']', '\\',
+    '#', ';', '\'', '`', ',', '.', '/', ' ', ' ', ' '};
+
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+  char name = ' ';
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+  if (keycode < 60) {
+    name = code_to_name[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
+           record->event.key.row, record->event.key.col,
+           keycode, name);
+}
+
+void oled_render_keylog(void) {
+    oled_write(keylog_str, false);
+}
+
+void render_bootmagic_status(bool status) {
+    /* Show Ctrl-Gui Swap options */
+    static const char PROGMEM logo[][2][3] = {
+        {{0x97, 0x98, 0}, {0xb7, 0xb8, 0}},
+        {{0x95, 0x96, 0}, {0xb5, 0xb6, 0}},
+    };
+    if (status) {
+        oled_write_ln_P(logo[0][0], false);
+        oled_write_ln_P(logo[0][1], false);
+    } else {
+        oled_write_ln_P(logo[1][0], false);
+        oled_write_ln_P(logo[1][1], false);
+    }
+}
+
+void oled_render_logo(void) {
+    static const char PROGMEM crkbd_logo[] = {
+        0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, 0x90, 0x91, 0x92, 0x93, 0x94,
+        0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, 0xb0, 0xb1, 0xb2, 0xb3, 0xb4,
+        0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, 0xd0, 0xd1, 0xd2, 0xd3, 0xd4,
+        0};
+    oled_write_P(crkbd_logo, false);
+}
+
+bool oled_task_user(void) {
+    if (is_keyboard_master()) {
+	oled_write_ln_P(PSTR("nice cock"), false);
+        oled_render_layer_state();
+        // oled_render_keylog();
+    } else {
+        oled_render_logo();
+    }
+    return false;
+}
+
+#endif // OLED_ENABLE
+
+void keyboard_post_init_user(void) {
+  config.raw = eeconfig_read_user();
+  layer_on(_DVORAK);
+}
```

Comment:
No need for this, actually.

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

This will handle setting the default layer, and setting it in eeprom.

Reply:
I ran into some problems with this. 
I use _QWERTY layer for russian and _DVORAK layer for english. 
I want _QWERTY layer be the base one and I want to have _DVORAK layer on top of it.
So when I type in russian on _QWERTY layer I will be able to turn on _DVORAK layer for Ctrl + C and other shortcuts